packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-target-group.ts

```diff
@@ -75,25 +77,35 @@ export class ApplicationTargetGroup extends TargetGroupBase implements IApplicat
   private readonly connectableMembers: ConnectableMember[];
   private readonly listeners: IApplicationListener[];
 
-  constructor(scope: Construct, id: string, props: ApplicationTargetGroupProps) {
-    const [protocol, port] = determineProtocolAndPort(props.protocol, props.port);
+  constructor(scope: Construct, id: string, props?: ApplicationTargetGroupProps) {
 
-    super(scope, id, props, {
+    const targetType = props ? props.targetType : undefined;
```

Comment:
Why introduce all these local variables?

Reply:
TBH I am not that proficient in TS and didn't really know what was idiomatic, since `props` had to be optional.
I tried to make it a little simpler now. Any advice on how to do it more idiomatic is greatly appreciated.